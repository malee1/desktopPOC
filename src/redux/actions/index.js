// eslint-disable-next-line import/prefer-default-export
export const getCustomer = (custObject) => {
  return {
    type: 'GETCUSTOMER',
    payload: custObject
  };
};

export const setIFrame = (url) => {
  return {
    type: 'SETIFRAME',
    payload: url
  };
};

export const getUrls = (urls) => {
  return {
    type: 'GETURLS',
    payload: urls
  };
};

export const getUser = (user) => {
  return {
    type: 'GETUSER',
    payload: user
  };
};

export const setAccessibility = (accessibility) => {
  return {
    type: 'SETACCESSIBILITY',
    payload: accessibility
  };
};
