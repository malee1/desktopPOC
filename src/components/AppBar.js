import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import { Button, ButtonGroup, Typography } from '@material-ui/core';
import { setIFrame } from '../redux/actions';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    padding: '20px'
  },
  buttonGroup: {
    display: 'flex'
    // flexDirection: 'column'
  },
  button: {
    margin: '10px'
  }
});

export default function AppBar() {
  const classes = useStyles();

  const buttonOne = useSelector((state) => state.urls.urls.urlOne);
  const buttonTwo = useSelector((state) => state.urls.urls.urlTwo);
  const buttonThree = useSelector((state) => state.urls.urls.urlThree);

  const dispatch = useDispatch();

  const handleButtonOneClick = () => {
    dispatch(setIFrame(`${buttonOne.url}`));
  };

  const handleButtonTwoClick = () => {
    dispatch(setIFrame(`${buttonTwo.url}`));
  };

  const handleButtonThreeClick = () => {
    dispatch(setIFrame(`${buttonThree.url}`));
  };

  return (
    <div className={classes.root} data-testid="app-bar">
      <Typography>App Bar</Typography>
      <ButtonGroup className={classes.buttonGroup} orientation="vertical" color="primary">
        <Button
          className={classes.button}
          variant="contained"
          color="primary"
          onClick={handleButtonOneClick}
          data-testid="button"
        >
          {buttonOne.name}
        </Button>
        <Button
          className={classes.button}
          variant="contained"
          color="primary"
          onClick={handleButtonTwoClick}
          data-testid="button"
        >
          {buttonTwo.name}
        </Button>
        <Button
          className={classes.button}
          variant="contained"
          color="primary"
          onClick={handleButtonThreeClick}
          data-testid="button"
        >
          {buttonThree.name}
        </Button>
      </ButtonGroup>
    </div>
  );
}
