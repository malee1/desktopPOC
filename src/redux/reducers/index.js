import { combineReducers } from 'redux';
import getCustomer from './getCustomer';
import getUrl from './getURL';

const rootReducer = combineReducers({
  customer: getCustomer,
  iFrame: getUrl
});

export default rootReducer;
