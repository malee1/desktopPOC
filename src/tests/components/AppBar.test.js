import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from '@testing-library/react';
import { setIFrame } from '../../redux/actions';

import AppBar from '../../components/AppBar';

const mockStore = configureMockStore();

const store = mockStore({
  user: {
    data: {
      fileId: '123456'
    }
  },
  urls: {
    urls: {
      urlOne: {
        name: 'test1',
        url: 'testUrlOne'
      },
      urlTwo: {
        name: 'test2',
        url: 'testUrlTwo'
      },
      urlThree: {
        name: 'test3',
        url: 'testUrlThree'
      }
    }
  },
  accessibility: {
    accessibility: false
  }
});

const store2 = mockStore({
  user: {
    data: {
      fileId: '567890',
      roleProfile: 'Alpha',
      boltOn: 'Accessibility'
    }
  },
  urls: {
    urls: {
      urlOne: {
        name: 'test1',
        url: 'testUrlOne'
      },
      urlTwo: {
        name: 'test2',
        url: 'testUrlTwo'
      },
      urlThree: {
        name: 'test3',
        url: 'testUrlThree'
      }
    }
  },
  accessibility: {
    accessibility: false
  }
});

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

describe('panel renders correctly', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AppBar />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('it renders the right number of buttons', () => {
  it('displays three buttons', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <AppBar />
      </Provider>
    );
    const buttons = getAllByTestId('button');
    expect(buttons.length).toBe(3);
  });
  it('displays the right text on the button', () => {
    const { getAllByText } = render(
      <Provider store={store}>
        <AppBar />
      </Provider>
    );
    const buttonOne = getAllByText('test1');
    const buttonTwo = getAllByText('test2');
    const buttonThree = getAllByText('test3');
    expect(buttonOne.length).toBe(1);
    expect(buttonTwo.length).toBe(1);
    expect(buttonThree.length).toBe(1);
  });
});

store.dispatch = jest.fn();

describe('it dispatches actions on click', () => {
  it('dispatches setIFrame on button 1 click', () => {
    const { getByText } = render(
      <Provider store={store}>
        <AppBar />
      </Provider>
    );
    const button = getByText('test1');
    fireEvent.click(button);
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(setIFrame('testUrlOne'));
  });
  it('dispatches setIFrame on button 2 click', () => {
    const { getByText } = render(
      <Provider store={store}>
        <AppBar />
      </Provider>
    );
    const button = getByText('test2');
    fireEvent.click(button);
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(setIFrame('testUrlTwo'));
  });
  it('dispatches setIFrame on button 3 click', () => {
    const { getByText } = render(
      <Provider store={store}>
        <AppBar />
      </Provider>
    );
    const buttonOne = getByText('test3');
    fireEvent.click(buttonOne);
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(setIFrame('testUrlThree'));
  });
});

describe('it shows the accessibility button if the bolt on is present on the user profile', () => {
  it('displays the button correctly', () => {
    const { getAllByTestId } = render(
      <Provider store={store2}>
        <AppBar />
      </Provider>
    );
    const button = getAllByTestId('button-accessibility');
    expect(button.length).toBe(1);
  });
});

describe('The button displays the accessibility modal', () => {
  it('displays the button correctly', () => {
    const { getAllByTestId, getByText } = render(
      <Provider store={store2}>
        <AppBar />
      </Provider>
    );
    const button = getByText('Display');
    fireEvent.click(button);
    const modal = getAllByTestId('accessibility-menu');
    expect(modal.length).toBe(1);
  });
});
