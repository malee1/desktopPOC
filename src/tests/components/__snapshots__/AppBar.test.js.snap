// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel renders correctly matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="app-bar"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    App Bar
  </p>
  <div
    className="MuiButtonGroup-root makeStyles-buttonGroup-2 MuiButtonGroup-vertical"
    role="group"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlinedPrimary makeStyles-button-3 MuiButton-containedPrimary"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        test1
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlinedPrimary makeStyles-button-3 MuiButton-containedPrimary"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        test2
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlinedPrimary makeStyles-button-3 MuiButton-containedPrimary"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        test3
      </span>
    </button>
  </div>
</div>
`;
